interfaces/portal/src/app/components/confirmation-dialog/confirmation-dialog.component.ts

```diff
@@ -46,16 +48,31 @@ export class ConfirmationDialogComponent<TMutationData = unknown> {
 
   readonly confirmDialog = viewChild.required<ConfirmDialog>('confirmDialog');
 
+  private event: TrackingEvent | undefined = undefined;
+
   askForConfirmation({
     resetMutation = true,
-  }: { resetMutation?: boolean } = {}) {
+    event = undefined,
+  }: {
+    resetMutation?: boolean;
+    event?: TrackingEvent;
+  } = {}) {
     this.confirmationService.confirm({});
+    if (event) {
+      this.event = event;
+    }
+
     if (resetMutation) {
       this.mutation().reset();
     }
   }
 
   onProceed() {
+    if (this.event) {
+      this.trackingService.trackEvent({ ...this.event });
```

Comment:
Wouldn't `...trackEven(this.event)` work? Or is there a TypeScript complaint then?